packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -966,10 +976,13 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 							);
 						}
 					}
-					await Promise.all(attachBlobsP);
+					await Promise.race([Promise.all(attachBlobsP), abortPromise]).catch(() => {});
```

Comment:
rather than being something we only monitor externally to the individual blob attaches, it might be better to integrate it inside such that we reject the blobP for any incomplete blobs on cancelation. 

i also wonder if we can re-order the operation done for each blob, that such that we only send the blob op after/during attachment. we'd need tests to see how that works. this would solve the other problem, as there would be no ops sent for unattached blobs.

Reply:
Reordering sending the blob op on attachment seems plausible. It could help with the problem of collecting detached blobs. 
Not sure I get what you mean in the first paragraph though, but the fact that we must resolve the blob handle first in order to attach it makes really hard to try to have control over resolve/reject blobs.